pulser-core/pulser/sequence/sequence.py

```diff
@@ -822,29 +830,35 @@ def enable_eom_mode(
             channel_obj.validate_pulse(on_pulse)
             amp_on = cast(float, amp_on)
             detuning_on = cast(float, detuning_on)
-
-            off_options = cast(
-                RydbergEOM, channel_obj.eom_config
-            ).detuning_off_options(amp_on, detuning_on)
-
+            eom_config = cast(RydbergEOM, channel_obj.eom_config)
             if not isinstance(optimal_detuning_off, Parametrized):
-                closest_option = np.abs(
-                    off_options - optimal_detuning_off
-                ).argmin()
-                detuning_off = off_options[closest_option]
+                detuning_off = eom_config.calculate_detuning_off(
+                    amp_on, detuning_on, optimal_detuning_off
+                )
                 off_pulse = Pulse.ConstantPulse(
                     channel_obj.min_duration, 0.0, detuning_off, 0.0
                 )
                 channel_obj.validate_pulse(off_pulse)
 
             if not self.is_parametrized():
+                phase_drift_params = _PhaseDriftParams(
+                    drift_rate=-detuning_off, ti=self.get_duration(channel)
+                )
                 self._schedule.enable_eom(
                     channel, amp_on, detuning_on, detuning_off
                 )
+                if correct_phase_drift:
+                    buffer_slot = self._last(channel)
+                    drift = phase_drift_params.calc_phase_drift(buffer_slot.tf)
+                    self._phase_shift(
+                        -drift, *buffer_slot.targets, basis=channel_obj.basis
```

Comment:
Why is there a minus in front of drift ? I thought you were taking `-detuning_off` in `phase_drift_params` because that's this is the correcting phase drift... (at the moment I have the feeling you could have a + in front of drift here and in front of detuning_off in _PhaseDriftParams and the results would be the same)

Reply:
As it is the same in disable_eom_mode, I definitely have the feeling there is something I don't understand 😅

Reply:
Ah yes, I struggled with this a bit at first too. Remember that the Bloch sphere convention we follow has the z-component of the Bloch rotation vector be -∂(t). This means that, in free precession, the accumalated phase is indeed -∂*t (for constant ∂). Then, to correct for this, you apply a phase shift in the opposite direction. You are right when you say that we could just get rid of the minus signs, but I figured this would be more explicit.

Reply:
Thanks for the explanation ! I had the -∂*t but I was mixing things up afterwards :) I think what puzzled me a bit was having a negative rate, but that's okay let's keep it this way.